azurerm/resource_arm_storage_share_migration.go

```diff
@@ -3,62 +3,76 @@ package azurerm
 import (
 	"fmt"
 	"log"
+	"strings"
 
-	"github.com/hashicorp/terraform/terraform"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
 	"github.com/tombuildsstuff/giovanni/storage/2018-11-09/file/shares"
 )
 
-func resourceStorageShareMigrateState(v int, is *terraform.InstanceState, meta interface{}) (*terraform.InstanceState, error) {
-	switch v {
-	case 0:
-		log.Println("[INFO] Found AzureRM Storage Share State v0; migrating to v1")
-		return migrateStorageShareStateV0toV1(is)
-	case 1:
-		log.Println("[INFO] Found AzureRM Storage Share State v1; migrating to v2")
-		return migrateStorageShareStateV1toV2(is, meta)
-	default:
-		return is, fmt.Errorf("Unexpected schema version: %d", v)
+// the schema schema was used for both V0 and V1
+func resourceStorageShareStateResourceV0V1() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateArmStorageShareName,
```

Comment:
Nit: I believe you can remove extraneous elements like `ValidateFunc`, `ForceNew`, `Required`, etc. from these old schema definitions since the state upgraders only need the type information, but either which way though!